examples/CMakeLists.txt

```diff
@@ -0,0 +1,79 @@
+# Minimum CMake required

```

Comment:
Adding a CMakeLists.txt file directly in the relevant directory is a good practice, but is counter to the pattern currently used in the rest of the project. @xfxyjwf If you're ok with spreading cmake files out within the rest of the repository, we can do this. Otherwise all this logic should be moved to cmake/examples.cmake